File: ts/1152/1051.ts
function heightChecker(heights: number[]): number {
  const n = 101
  const bucket: number[] = new Array(n).fill(0)
  let cnt = 0
  for (const h of heights) {
    bucket[h] += 1
  }
  for (let i = 1, j = 0; i < n; ++i) {
    while (bucket[i] > 0) {
      if (i !== heights[j]) cnt += 1
      j += 1
      bucket[i] -= 1
    }
  }
  return cnt
};

File: py3/1152/1051.py
from typing import List

class Solution:
    def heightChecker(self, heights: List[int]) -> int:
        n = 101
        buckets = [0] * n
        for v in heights:
            buckets[v] += 1
        i, cnt = 0, 0
        for j in range(1, n):
            while buckets[j] > 0:
                if j != heights[i]:
                    cnt += 1
                i += 1
                buckets[j] -= 1
        return cnt